From de6a62d5369bad2c32cdf8735b5195c857a8345f Mon Sep 17 00:00:00 2001
From: Mike Sul <mike.sul@foundries.io>
Date: Thu, 15 Jun 2023 17:48:41 +0200
Subject: [PATCH] Increase TCP dialer timeouts

Increase the TCP dialer timeouts that allegedly is supposed to minimize
occurrence of TCP connection related errors/timeouts, including DNS
resolution.

Upstream-Status: Inappropriate [lmp specific]

Signed-off-by: Mike <mike.sul@foundries.io>
---
 pkg/tlsclientconfig/tlsclientconfig.go | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/vendor/github.com/containers/image/v5/pkg/tlsclientconfig/tlsclientconfig.go b/vendor/github.com/containers/image/v5/pkg/tlsclientconfig/tlsclientconfig.go
index 285203ba..ccea2c70 100644
--- a/vendor/github.com/containers/image/v5/pkg/tlsclientconfig/tlsclientconfig.go
+++ b/vendor/github.com/containers/image/v5/pkg/tlsclientconfig/tlsclientconfig.go
@@ -91,14 +91,14 @@ func hasFile(files []os.DirEntry, name string) bool {
 // NewTransport Creates a default transport
 func NewTransport() *http.Transport {
 	direct := &net.Dialer{
-		Timeout:   30 * time.Second,
-		KeepAlive: 30 * time.Second,
+		Timeout:   60 * time.Second,
+		KeepAlive: 60 * time.Second,
 		DualStack: true,
 	}
 	tr := &http.Transport{
 		Proxy:               http.ProxyFromEnvironment,
 		DialContext:         direct.DialContext,
-		TLSHandshakeTimeout: 10 * time.Second,
+		TLSHandshakeTimeout: 30 * time.Second,
 		// TODO(dmcgowan): Call close idle connections when complete and use keep alive
 		DisableKeepAlives: true,
 	}
-- 
2.40.1

